content/articles/mock-a-datasource-for-react-using-json-server/index.md

```diff
@@ -0,0 +1,837 @@
+
+```shell
+npx create-react-app react-budget-tracker
+```
+
+#### New project structure
+- Create the following application folder structure:
+
+```shell
+.
+├── node_modules
+├── public
+├── src
+│   └── components
+│       ├── About.js
+│       ├── AddBudget.js
+│       ├── Budget.js
+│       ├── Budgets.js
+│       ├── Button.js
+│       ├── Footer.js
+│       └── Header.js
+│   ├── App.js
+│   ├── index.css
+│   ├── index.js
+│   └── reportWebVitals.js
+├── .eslintcache
+├── .gitignore
+├── db.json
+├── package.json
+├── package-lock.json
+├── README.md
+└── yarn.lock
+```
+
+#### Header component
+- In the 'Header.js', copy and paste the code below:
+
+```js
+import PropTypes from 'prop-types'
+import { useLocation } from 'react-router-dom'
+import Button from './Button'
+
+const Header = ({ title, onAdd, showAdd }) => {
+  const location = useLocation()
+
+  return (
+    <header className='header'>
+      <h1>{title}</h1>
+      {location.pathname === '/' && (
+        <Button
+          color={showAdd ? 'red' : 'green'}
+          text={showAdd ? 'Close' : 'Add'}
+          onClick={onAdd}
+        />
+      )}
+    </header>
+  )
+}
+
+Header.defaultProps = {
+  title: 'Budget Tracker',
+}
+
+Header.propTypes = {
+  title: PropTypes.string.isRequired,
+}
+
+export default Header
+```
+
+This code creates a simple header for a web page and allows it for use.
+
+#### Footer component
+- In the 'Footer.js' file, the following will be the code:
+
+```js
+import { Link } from 'react-router-dom'
+
+const Footer = () => {
+  return (
+    <footer>
+      <p>Copyright &copy; 2021</p>
+      <Link to='/about'>About</Link>
+    </footer>
+  )
+}
+
+export default Footer
+```
+
+#### Button component
+- Add a button that is a reusable component for multiple click actions. For example, this button will open the Add form and close the form. The code for the button is as shown below:
+
+```js
+import PropTypes from 'prop-types'
+
+const Button = ({ color, text, onClick }) => {
+  return (
+    <button
+      onClick={onClick}
+      style={{ backgroundColor: color }}
+      className='btn'
+    >
+      {text}
+    </button>
+  )
+}
+
+Button.defaultProps = {
+  color: 'steelblue',
+}
+
+Button.propTypes = {
+  text: PropTypes.string,
+  color: PropTypes.string,
+  onClick: PropTypes.func,
+}
+
+export default Button
+```
+
+#### About page component
+- Now, add a webpage that may contain other page contents. In this case, an About page. This page's code is shown below:
+
+```js
+import { Link } from 'react-router-dom'
+
+const About = () => {
+  return (
+    <div>
+      <h4>Version 1.0.0</h4>
+      <Link to='/'>
+          <button type="button" className={"btn btn-secondary"}>
+              Go Back
+          </button>
+      </Link>
+    </div>
+  )
+}
+
+export default About
+```
+
+It contains a button to go back to the previous page.
+
+#### Budget.js file
+- In the 'Budget.js' file, the following will be its code:
```

Comment:
What is this file for? What does it contain?

Don't just specify the file name. The reader wouldn't be able to understand your thought process, just with the filename

Reply:
Do this for other code snippets as well